docs/docs/guides/evaluation/scorers.md

```diff
@@ -0,0 +1,318 @@
+
+---
+
+### `SummarizationScorer`
+
+Use an LLM to compare a summary to the original text and evaluate the quality of the summary.
+
+```python
+from weave.scorers import SummarizationScorer
+
+llm_client = # initialize your LLM client here
+
+scorer = SummarizationScorer(
+    client=llm_client, 
+    model_id="gpt4o"
+)
+```
+
+**How It Works:**
+
+This scorer evaluates summaries in two ways:
+
+1. **Entity Density:** Checks the ratio of unique entities (like names, places, or things) mentioned in the summary to the total word count in the summary in order to estimate the "information density" of the summary. Uses an LLM to extract the entities. Similar to how entity density is used in the Chain of Density paper, https://arxiv.org/abs/2309.04269
+
+2. **Quality Grading:** Uses an LLM-evaluator to grade the summary as `poor`, `ok`, or `excellent`. These grades are converted to scores (0.0 for poor, 0.5 for ok, and 1.0 for excellent) so you can calculate averages.
+
+**Customization:**
+- Adjust `summarization_evaluation_system_prompt` and `summarization_evaluation_prompt` to define what makes a good summary.
+
+**Notes:**
+- This scorer uses the `InstructorLLMScorer` class.
+- The `score` method expects the original text that was summarized to be present in the `input` column of the dataset. Use the `column_map` class attribute to map `input` to the correct dataset column if needed.
+
+
+---
+
+### `OpenAIModerationScorer`
+
+The `OpenAIModerationScorer` uses OpenAI's Moderation API to check if the AI system's output contains disallowed content, such as hate speech or explicit material.
+
+```python
+from weave.flow.scorers.moderation_scorer import OpenAIModerationScorer
+import openai
+
+oai_client = OpenAI(api_key=...) # initialize your LLM client here
+
+scorer = OpenAIModerationScorer(
+    client=oai_client,
+    model_id="text-embedding-3-small"
+)
+```
+
+**How It Works:**
+
+- Sends the AI's output to the OpenAI Moderation endpoint and returns a dictionary indicating whether the content is flagged and details about the categories involved.
+
+**Notes:**
+- Requires the `openai` Python package.
+- The client must be an instance of OpenAI's `OpenAI` or `AsyncOpenAI` client.
+
+---
+
+### `EmbeddingSimilarityScorer`
+
+The `EmbeddingSimilarityScorer` computes the cosine similarity between the embeddings of the AI system's output and a target text from your dataset. It's useful for measuring how similar the AI's output is to a reference text.
+
+```python
+from weave.flow.scorers.similarity_score import EmbeddingSimilarityScorer
+
+llm_client = ...  # initialise your LlM client
+
+similarity_scorer = EmbeddingSimilarityScorer(
+    client=llm_client
+    target_column="reference_text",  # the dataset column to compare the output against
+    threshold=0.4  # the cosine similarity threshold to use  
```

Comment:
Not everyone will know what this is.  It might be helpful to give some context for what a reasonable range is, or how to think about setting this threshold

Reply:
In future this may also be a good place to link to resources like an arxiv paper or some research we've done internally